test/excluded_prefixes_collector_test.go

```diff
@@ -0,0 +1,228 @@
+}
+
+func (et *ExcludedPrefixesSuite) TestWithAllPrefixes() {
+	envPrefixes := []string{
+		"127.0.0.0/8",
+		"134.65.0.0/16",
+	}
+
+	et.Require().NoError(k8s.ApplyDaemonSet("../deployments/prefixes-collector/collector.yaml", func(collector *v1.DaemonSet) {
+		collector.Spec.Template.Spec.Containers[0].Env = append(collector.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  excludedPrefixesEnv,
+			Value: strings.Join(envPrefixes, ","),
+		})
+	}))
+
+	et.Require().NoError(exechelper.Run("kubectl apply -f ./files/userConfigMap.yaml", et.options...))
+	defer func() {
+		et.Require().NoError(exechelper.Run("kubectl delete configmaps excluded-prefixes-config", et.options...))
+	}()
+
+	expectedPrefixes := append(append(kubeAdmPrefixes, envPrefixes...), userConfigPrefixes...)
+	et.Eventually(et.checkPrefixes(expectedPrefixes), time.Second*30, time.Second)
+}
+
+func (et *ExcludedPrefixesSuite) TestWithCorrectEnvPrefixes() {
+	envPrefixes := []string{
+		"127.0.0.0/8",
+		"134.65.0.0/16",
+	}
+
+	et.Require().NoError(k8s.ApplyDaemonSet("../deployments/prefixes-collector/collector.yaml", func(collector *v1.DaemonSet) {
+		collector.Spec.Template.Spec.Containers[0].Env = append(collector.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  excludedPrefixesEnv,
+			Value: strings.Join(envPrefixes, ","),
+		})
+	}))
+
+	expectedPrefixes := append(kubeAdmPrefixes, envPrefixes...)
+	et.Eventually(et.checkPrefixes(expectedPrefixes), time.Second*30, time.Second)
+}
+
+func (et *ExcludedPrefixesSuite) TestWithIncorrectEnvPrefixes() {
+	envPrefixes := []string{
+		"306.306.306.0",
+	}
+
+	et.Require().NoError(k8s.ApplyDaemonSet("../deployments/prefixes-collector/collector.yaml", func(collector *v1.DaemonSet) {
+		collector.Spec.Template.Spec.Containers[0].Env = append(collector.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  excludedPrefixesEnv,
+			Value: strings.Join(envPrefixes, ","),
+		})
+	}))
+
+	et.Eventually(func() bool {
+		podInfo, err := k8s.GetPod(prefixesFileName, "", map[string]string{
+			"app": "excluded-prefixes-collector",
+		})
+		et.Require().NoError(err)
+		return podInfo == nil
+	}, time.Second*15, time.Second)
+}
+
+func TestExcludedPrefixesSuite(t *testing.T) {
+	suite.Run(t, &ExcludedPrefixesSuite{})
+}
+
+func (et *ExcludedPrefixesSuite) checkPrefixes(expectedPrefixes []string) func() bool {
+	var podInfo *corev1.Pod
+	dirName := os.TempDir()
+	localPrefixesPath := filepath.Join(dirName, prefixesFileName)
+	isStarted := false
+	labels := map[string]string{
+		"app": "alpine",
+	}
+
+	return func() bool {
+		if !isStarted {
+			var err error
+			podInfo, err = k8s.GetPod(collectorNamespace, "", labels)
+			et.Require().NoError(err)
+
+			isStarted = podInfo != nil && podInfo.Status.Phase == corev1.PodRunning
+			if !isStarted {
+				return false
+			}
+		}
+
+		kubeCpCmd := fmt.Sprintf("kubectl cp -n excluded-prefixes-collector %v:%v %v", podInfo.Name, nsmConfigDir, dirName)
```

Comment:
Can we check the source of the file via `kubectl exec -ti $POD_NAME -- cat $PATH_TO_FILE`?

Reply:
Replaced with `cat`